llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -8663,21 +8678,21 @@ std::optional<bool> llvm::isImpliedCondition(const Value *LHS, const Value *RHS,
     if (std::optional<bool> Imp =
             isImpliedCondition(LHS, RHS1, DL, LHSIsTrue, Depth + 1))
       if (*Imp == true)
-        return true;
+        return !InvertRHS;
```

Comment:
Oh, it looks incorrect :(

Reply:
Its confusing enough I think it deserves a comment.

Reply:
NB: I don't any of your tests actually go through this control flow

Reply:
Just a heads up, im working on a follow up patch to this to handle more of the `and/or` cases + more relationships in general (LMK if you are so we can avoid duplicating work).